ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ValueNetsingIT.scala

```diff
@@ -0,0 +1,116 @@
+
+  @tailrec
+  def toLong(n: Nat, acc: Long = 0): Int =
+    n match {
+      case Nat.Z(_) => 0
+      case Nat.S(n) => toLong(n, acc + 1)
+    }
+
+  def reify[X](future: Future[X])(implicit ec: ExecutionContext): Future[Either[Throwable, X]] =
+    future.transform(x => Success(x.toEither))
+
+  List[Long](30, 100, 101, 110, 200).foreach { depth =>
+    val accepted = depth <= 100
+    val result = if (accepted) "Accept" else "Reject"
+
+    def test[T](description: String)(
+        update: ExecutionContext => (
+            ParticipantTestContext,
+            Party,
+        ) => Future[Either[Throwable, T]]
+    ) =
+      super.test(
+        s"${result}$description$depth",
+        s"${result.toLowerCase}s $description with of $depth",
+        allocate(SingleParty),
+      )(implicit ec => { case Participants(Participant(alpha, party)) =>
+        update(ec)(alpha, party).map {
+          case Right(_) if accepted => ()
+          case Left(err: Throwable) if !accepted =>
+            assertGrpcError(err, Status.Code.INVALID_ARGUMENT, None)
+          case otherwise => fail("Unexpected " + otherwise.fold(_ => "success", _ => "failure"))
+        }
+      })
+
+    test("CreateArgument") { implicit ec => (alpha, party) =>
+      reify(alpha.create(party, Contract(party, depth, toNat(depth - 2))))
+
+    }
+
+    test("ExerciseArgument") { implicit ec => (alpha, party) =>
+      for {
+        handler <- alpha.create(party, Handler(party))
+        result <- reify(alpha.exercise(party, handler.exerciseDestruct(_, toNat(depth - 2))))
+      } yield result
+    }
+
+    test("ExerciseOutput") { implicit ec => (alpha, party) =>
+      for {
+        handler <- alpha.create(party, Handler(party))
+        result <- reify(alpha.exercise(party, handler.exerciseConstruct(_, depth - 2)))
+      } yield result
+    }
+
+    test("Create") { implicit ec => (alpha, party) =>
+      for {
+        handler <- alpha.create(party, Handler(party))
+        result <- reify(alpha.exercise(party, handler.exerciseCreate(_, depth - 2)))
+      } yield result
+    }
+
+    test("CreateKey") { implicit ec => (alpha, party) =>
+      for {
+        handler <- alpha.create(party, Handler(party))
+        result <- reify(alpha.exercise(party, handler.exerciseCreateKey(_, depth - 2)))
+      } yield result
+    }
+
+    if (accepted)
+      test("FetchByKey") { implicit ec => (alpha, party) =>
+        for {
+          handler <- alpha.create(party, Handler(party))
+          _ <- alpha.exercise(party, handler.exerciseCreateKey(_, depth - 2))
+          result <- reify(alpha.exercise(party, handler.exerciseFetchByKey(_, depth - 2)))
+        } yield result
+      }
+
+    test("FailingLookupByKey") { implicit ec => (alpha, party) =>
+      for {
+        handler <- alpha.create(party, Handler(party))
+        result <- reify(alpha.exercise(party, handler.exerciseLookupByKey(_, depth - 2)))
+      } yield result
+    }
+
+    if (accepted)
```

Comment:
well I guess the one above wasn’t the only one after all :) I’m still confused as to what makes those tests special

Reply:
Similarly, no point to test a successful lookup by key, when we actually cannot create the contract to be look up for.